src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -180,8 +186,7 @@ public Task StopAsync()
             {
                 Log.UnexpectedExceptionInStop(Logger, ConnectionId, ex);
             }
-
-            return Task.CompletedTask;
+            await _serviceConnectionEndTcs.Task;
```

Comment:
Add a case to test if `StopAsync` could quit successfully when it failed to start.